Tests/AnnoDesigner.Tests/AdjecentCellGrouperTests.cs

```diff
@@ -0,0 +1,248 @@
+﻿using System.IO.Abstractions.TestingHelpers;
+using System.Linq;
+using System.Windows;
+using AnnoDesigner.Helper;
+using Xunit;
+
+namespace AnnoDesigner.Tests
+{
+    public class AdjacentCellGrouperTests
+    {
+        private class Item
+        {
+            public int Value { get; set; }
+        }
+
+        private AdjacentCellGrouper grouper = new AdjacentCellGrouper();
+
+        /// <summary>
+        /// Returns grid representation of string lines.
+        /// Character "X" is used to represent grid cell being occupied. Any other character means empty cell.
+        /// Start of coordinates is in top left corner, width is horizontal and height is vertical.
+        /// </summary>
+        /// <example>
+        /// "XX  ",
+        /// "XX  ",
+        /// "XXXX",
+        /// "XXXX"
+        /// </example>
+        public static bool[][] ParseGrid(params string[] gridLines)
+        {
+            var preTranspose = gridLines.Select(line => line.Select(c => c == 'X').ToArray()).ToArray();
+            var postTranspose = Enumerable.Range(0, gridLines.Max(i => i.Length)).Select(i => new bool[gridLines.Length]).ToArray();
+
+            for (var i = 0; i < gridLines.Length; i++)
+                for (var j = 0; j < gridLines[i].Length; j++)
+                    postTranspose[j][i] = preTranspose[i][j];
+
+            return postTranspose;
+        }
+
+        [Fact]
+        public void WholeArray()
```

Comment:
Could we rename these tests so they are in line with rest of the tests in the codebase? The usual pattern we're following is <method name>_<test conditions>_<expected result>, just look at some other tests for reference. The sections can be optional if it doesn't make sense - its not a hard and fast rule.

Basically, you want to be able to read the test from the test runner and know what its doing. If I read `WholeArray`, that doesn't explain anything to me. If I read `GroupAdjacentCells_SquareArrayNoGaps_ReturnsSingleResult`, then I know exactly what the test is actually doing.